addons/mail/models/mail_channel.py

```diff
@@ -190,7 +190,7 @@ def create(self, vals_list):
             if any(cmd[0] not in (4, 6) for cmd in partner_ids_cmd):
                 raise ValidationError(_('Invalid value when creating a channel with members, only 4 or 6 are allowed.'))
             partner_ids = [cmd[1] for cmd in partner_ids_cmd if cmd[0] == 4]
-            partner_ids += [cmd[2] for cmd in partner_ids_cmd if cmd[0] == 6]
+            partner_ids += [pid for cmd in partner_ids_cmd if cmd[0] == 6 for pid in cmd[2]]
```

Comment:
This change should be discussed with the sm team.

Reply:
We probably don't want it indeed. Isn't it possible to use the expected format instead?

Or even better pass members to `create_group`, or to `add_members`?

Reply:
Don't use magic numbers. Use the [defined constants](https://github.com/odoo/odoo/blob/52256398c9e6467e462846f9a54507423b8ca90f/odoo/fields.py#L3083-L3089)
```suggestion
            partner_ids += [pid for cmd in partner_ids_cmd if cmd[0] == fields.Command.SET for pid in cmd[2]]
```

Reply:
@seb-odoo indeed. We're using `add_members` now so this change is not necessary anymore. If you check the diff, I only updated the current code following @KangOl 's suggestion of using the constants :smile:

Reply:
Please don't make unnecessary diff, if you want to clean commands feel free to open a new PR to take care of it, but it shouldn't be done here.